io/net/htx_nic_devices.py

```diff
@@ -0,0 +1,591 @@
+        self.net_id2 = self.params.get("net_id2", '*', default=None)
+        self.mdt_file = self.params.get("mdt_file", '*', default="net.mdt")
+        self.time_limit = int(self.params.get("time_limit",
+                                              '*', default=2)) * 3600
+        self.query_cmd = "htxcmdline -query -mdt %s" % self.mdt_file
+
+    def login(self, ip, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        p = pxssh.pxssh()
+        # Work-around for old pxssh not having options= parameter
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'StrictHostKeyChecking=no'"
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'UserKnownHostsFile /dev/null' "
+        p.force_password = True
+
+        p.login(ip, username, password)
+        p.sendline()
+        p.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        p.set_unique_prompt()
+        p.prompt(timeout=60)
+        self.pxssh = p
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on lpar %s", command)
+        c = self.pxssh
+        c.sendline(command)
+        c.expect("\n")  # from us
+        c.expect(c.PROMPT, timeout=timeout)
+        output = c.before.splitlines()
+        c.sendline("echo $?")
+        c.prompt(timeout)
+        exitcode = int(''.join(c.before.splitlines()[1:]))
+        if exitcode != 0:
+            raise CommandFailed(command, output, exitcode)
+        return output
+
+    def test(self):
+        """
+        This test will be in two phases
+        Phase 1: Configure all necessary pre-setup steps for both the
+                 interfaces in both Host & Peer
+        Phase 2: Start the HTX setup & execution of test for a time_limit
+                 Monitor HTX error log for any errors in both Host & Peer
+        """
+        self.setup_htx_nic()
+        self.run_htx()
+
+    def setup_htx_nic(self):
+        self.update_host_peer_names()
+        self.generate_bpt_file()
+        self.check_bpt_file_existence()
+        self.update_otherids_in_bpt()
+        self.update_net_ids_in_bpt()
+        self.htx_configure_net()
+
+    def update_host_peer_names(self):
+        """
+        Update hostname & ip of both Host & Peer in /etc/hosts file of both
+        Host & Peer
+        """
+        host_name = process.system_output("hostname", ignore_status=True,
+                                          shell=True, sudo=True)
+        peer_name = self.run_command("hostname")[-1]
+        hosts_file = '/etc/hosts'
+        self.log.info("Updating hostname of both Host & Peer in \
+                      %s file", hosts_file)
+        with open(hosts_file, 'r') as file:
+            filedata = file.read()
+        file.close()
```

Comment:
The `with` already closes the `file`, please remove the unnecessary double-close.